src/target/target_probe.c

```diff
@@ -19,24 +19,39 @@
 
 #include "target_probe.h"
 
-#define CORTEXA_PROBE_WEAK_NOP __attribute__((weak, alias("cortexa_probe_nop")))
-#define CORTEXM_PROBE_WEAK_NOP __attribute__((weak, alias("cortexm_probe_nop")))
-#define TARGET_PROBE_WEAK_NOP __attribute__((weak, alias("target_probe_nop")))
+#ifdef __APPLE__
+// https://bugs.llvm.org/show_bug.cgi?id=17775
+#define APPLE_STATIC
+// https://gcc.gnu.org/onlinedocs/cpp/Pragmas.html
+#define DO_PRAGMA_(x) _Pragma(#x)
+#define DO_PRAGMA(x)  DO_PRAGMA_(x)
+// __attribute__((alias)) is not supported in AppleClang.
+#define weak_alias(name, aliasname)  DO_PRAGMA(weak name = aliasname)
+#define CORTEXA_PROBE_WEAK_NOP(name) weak_alias(name, cortexa_probe_nop)
+#define CORTEXM_PROBE_WEAK_NOP(name) weak_alias(name, cortexm_probe_nop)
+#define TARGET_PROBE_WEAK_NOP(name)  weak_alias(name, target_probe_nop)
+#else
+#define APPLE_STATIC static inline
+#define CORTEXA_PROBE_WEAK_NOP(name) \
+	extern bool name(ADIv5_AP_t *, uint32_t) __attribute__((weak, alias("cortexa_probe_nop")))
```

Comment:
Looks like these macros need ';' added to the end to mark the ends of the decls properly